lib/tmuxinator/cli.rb

```diff
@@ -164,16 +164,17 @@ def generate_project_file(name, path)
       end
 
       def create_project(project_options = {})
-        attach_opt = project_options[:attach]
-        attach = !attach_opt.nil? && attach_opt ? true : false
-        detach = !attach_opt.nil? && !attach_opt ? true : false
+        attach = detach = false
+        attach = true if project_options[:attach] == true
+        detach = true if project_options[:attach] == false
```

Comment:
Would this be better as

```ruby
attach = project_options[:attach]  # will be a boolean-ish string value
detach = !attach # will be an actual TrueClass/FalseClass
```

Reply:
This piece of code is more confusing that it should be because of how the `attach` option is overloaded. The `attach`/`detach` variables are mutually exclusive iff `project_options[:attach]` is not `nil`. Also, AFAIK, Thor coerces "f" and "false" to `false` and any other string or the empty flag to `true`, so I don't think there's any reason to cast. I'll add a comment which documents this.

FWIW, I had actually used your approach in my initial refactoring (which was prompted by Code Climate failing the build -- and the only reason I touched any of this) but realized that it didn't cover all of possible scenarios. There were also no tests covering these assignments, which is why I added them.

I'm thinking that we should introduce a top-level detach option. That would remove all of the conditional logic and simplify this method. I'll open a separate issue which addresses the possibility of adding the new option.